diff --git "a/cores/zxnext/src/ram/dpram2.vhd" "b/cores/zxnext/src/ram/dpram2.vhd"
index 6dd1c4d..29aba45 100644
--- "a/cores/zxnext/src/ram/dpram2.vhd"
+++ "b/cores/zxnext/src/ram/dpram2.vhd"
@@ -61,7 +61,8 @@ architecture rtl of dpram2 is
    type ram_t is array (0 to 2**addr_width_g-1) of bit_vector(data_width_g-1 downto 0);
 
    impure function InitRamFromFile (RamFileName : in string) return ram_t is
-      file     RamFile     : text is in RamFileName;
+      file     RamFile     : text;
+	  variable open_status :FILE_OPEN_STATUS;       
       variable RamFileLine : line;
       variable RAM         : ram_t;
       variable RAM_0       : ram_t := (others => (others => '0'));
@@ -69,12 +70,17 @@ architecture rtl of dpram2 is
       if RamFileName = "init/none.bin.txt" then
          return RAM_0;
       else
-         for I in ram_t'range loop
-            readline (RamFile, RamFileLine);
-            read (RamFileLine, RAM(I));
-         end loop;
-         return RAM;
-      end if;
+         file_open(open_status, RamFile, RamFileName, read_mode);
+         if open_status /= open_ok then
+            return RAM_0;
+         else
+            for I in ram_t'range loop
+               readline (RamFile, RamFileLine);
+               read (RamFileLine, RAM(I));
+            end loop;
+            return RAM;
+         end if;
+      end if;     
    end function;
 
    signal ram_q : ram_t := InitRamFromFile(init_file_g);
